Userland/Libraries/LibGUI/Calendar.cpp

```diff
@@ -271,7 +284,7 @@ void Calendar::update_tiles(unsigned view_year, unsigned view_month)
             view_month = i + 1;
 
         auto first_day_of_current_month = Core::DateTime::create(view_year, view_month, 1);
-        unsigned start_of_month = first_day_of_current_month.weekday();
+        unsigned start_of_month = (first_day_of_current_month.weekday() - m_first_day_of_week + 7) % 7;
```

Comment:
Does the `+ 7` change anything here?

Reply:
I believe so. `m_first_day_of_week` could be bigger than `first_day_of_current_month.weekday()` resulting in a negative value (or a wrap-around with `unsigned` values).